[S4SDK] Update Tutorials 2 & 3 for Java Cloud SDK on Cloud Foundry

Johannes-Schneider opened the pull request:
@MatKuhr @artemkovalyov 

### What has been done
* Update outdated screenshots of Cloud Foundry UI
* Replace screenshots of project structure with more maintainable tree-like code
* Update instructions on how to connect your first CF application to the SAP Business Hub sandbox service

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/SAPDocuments/Tutorials?pullRequest=6405) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/SAPDocuments/Tutorials?pullRequest=6405) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/SAPDocuments/Tutorials?pullRequest=6405) it.</sub>

MichaelCzcz reviewed (changes requested):
Hello Johannes,
there are some occurences of the so called royal "we" scattered acros both tutorials, where you are addressing the user with a task, but say something like: "In order to see what's happening, we have to do this and that.". According to the[ style guide](https://jam4.sapjam.com/wiki/show/MozDWaPczznL8NKeGEk0y9?_lightbox=true), this should not be done. Please you "you" unless you are referring to SAP.
This affects also parts that are not part of this pull request. Please rephrase those parts that are affected. Thank you.

Johannes-Schneider commented:
Hi Micheal,
thank you for your review and pointing these issues out. I went through our entire mission and replaced (hopefully) all corresponding occurrences of the 'royal we' with the appropriate 'you'.

@MatKuhr @artemkovalyov 
I also updated the CF screenshots and instructions in the second group of the mission.

artemkovalyov commented on tutorials/s4sdk-caching/s4sdk-caching.md:
```diff
@@ -41,7 +41,7 @@ Caching is applicable whenever:
 
 
-If these requirements apply to your use case, then we highly recommend that you use the caching features provided by the SAP Cloud SDK in your application. Now that we have seen why caching is useful, we'll tend to what the Cloud SDK provides to an application in that regard.
+If these requirements apply to your use case, then we highly recommend that you use the caching features provided by the SAP Cloud SDK in your application. Now that you have seen why caching is useful, you will learn what the Cloud SDK provides to an application in that regard.
```
```suggestion
If these requirements apply to your use case, then we highly recommend that you use the caching features provided by the SAP Cloud SDK in your application. Now that you have seen why caching is useful, you will learn what the Cloud SDK provides to enable it.
```

artemkovalyov commented on tutorials/s4sdk-caching/s4sdk-caching.md:
```diff
@@ -50,7 +50,7 @@ If these requirements apply to your use case, then we highly recommend that you
 The Cloud SDK makes it easy to cache your requests since it handles most of the complexity under the hood. This includes handling of tenant-aware requests, which is essential in a multi-tenant application. The SDK will isolate the cache on a tenant or principal level automatically, if your request requires it.
 
-In SAP Cloud SDK, `JCache` (`JSR 107`) is used as underlying caching technology. In this tutorial we will use the `JCache` adapter [Caffeine] (https://github.com/ben-manes/caffeine) for our purpose, but you can use any implementation you like. For Caffeine, add the following dependency to your application `pom.xml`:
+In SAP Cloud SDK, `JCache` (`JSR 107`) is used as underlying caching technology. In this tutorial you will use the `JCache` adapter [Caffeine] (https://github.com/ben-manes/caffeine) for this purpose, but you can use any implementation you like. For Caffeine, add the following dependency to your application `pom.xml`:
 
```
```suggestion
In SAP Cloud SDK, `JCache` (`JSR 107`) is used as underlying caching technology. In this tutorial, you will use the `JCache` adapter [Caffeine] (https://github.com/ben-manes/caffeine) for this purpose, but you can use any implementation you like. For Caffeine, add the following dependency to your application `pom.xml`:

```

artemkovalyov commented on tutorials/s4sdk-caching/s4sdk-caching.md:
```diff
@@ -83,14 +83,14 @@ final ResilienceConfiguration.CacheConfiguration cacheConfig =
 2. Declare the parameters that need to be stored together with the cached data
 
-The first step is obviously necessary, since we want to store the data for a limited amount of time. The longer we keep the cached information, the more outdated it will become. How long you want to keep data cached depends on your specific use case. How fast do you expect the information to be outdated? How frequently will the data be accessed? The timeout sets a trade-off between the data being ...
+The first step is obviously necessary, since data should be stored for a limited time only. The longer you keep the cached information, the more outdated it will become. How long you want to keep data cached depends on your specific use case. How fast do you expect the information to be outdated? How frequently will the data be accessed? The timeout sets a trade-off between the data being up to d ...
```
```suggestion
The first step is obviously necessary since data should be stored for a limited time only. The longer you keep the cached information, the more outdated it will become. How long you want to keep data cached depends on your specific use case. How fast do you expect the information to be outdated? How frequently will the data be accessed? The timeout sets a trade-off between the data is up to date and the application is responsive.
```

artemkovalyov commented on tutorials/s4sdk-caching/s4sdk-caching.md:
```diff
@@ -83,14 +83,14 @@ final ResilienceConfiguration.CacheConfiguration cacheConfig =
+The first step is obviously necessary, since data should be stored for a limited time only. The longer you keep the cached information, the more outdated it will become. How long you want to keep data cached depends on your specific use case. How fast do you expect the information to be outdated? How frequently will the data be accessed? The timeout sets a trade-off between the data being up to d ...
 
-Secondly, we specify the parameters that are to be cached with the data. For our request to retrieve a list of business partners no parameters are necessary, so we build our cache `wihtoutParameters`. But imagine you want to fetch information about a specific business partner by passing an ID to the system. In order to cache such a request the cache needs to not only remember the result received, ...
+Secondly, you specify the parameters that are to be cached with the data. For your request to retrieve a list of business partners no parameters are necessary, so you can build your cache `wihtoutParameters`. But imagine you want to fetch information about a specific business partner by passing an ID to the system. In order to cache such a request the cache needs to not only remember the result r ...
```
```suggestion
Secondly, you specify the parameters that are to be cached with the data. For your request to retrieve a list of business partners no parameters are necessary, so you can build your cache `wihtoutParameters`. But imagine you want to fetch information about a specific business partner by passing an ID to the system. In order to cache such a request, the cache needs to not only remember the result received, but also the ID that was associated with it. In such a case one can simply pass such parameters by using `.withParameters(param1, param2, ..)`.
```

artemkovalyov commented on tutorials/s4sdk-caching/s4sdk-caching.md:
````diff
@@ -154,7 +161,7 @@ public void testWithFallback() {
 ```
 
-Executed on it's own, this test will still pass as it did before. However, if `testService` is run prior to this test the result will be cached and `testWithFallback` will fail since the cached data is returned instead of falling back to an empty list. By changing the test we avoided such issues since our new test does not interfere with other tests. While being sufficient for this tutorial, in a ...
+Executed on it's own, this test will still pass as it did before. However, if `testService` is run prior to this test the result will be cached and `testWithFallback` will fail since the cached data is returned instead of falling back to an empty list. By changing the test you avoided such issues since your new test does not interfere with other tests. While being sufficient for this tutorial, in ...
````
```suggestion
Executed on its own, this test will still pass as it did before. However, if `testService` is run prior to this test the result will be cached and `testWithFallback` will fail since the cached data is returned instead of falling back to an empty list. By changing the test you avoided such issues since your new test does not interfere with other tests. While being sufficient for this tutorial, in a productive setting one should implement a more sophisticated and robust test setup, where caches are invalidated between tests.
```

artemkovalyov reviewed (changes requested):
Thanks for a great job! Check minor suggestions from my side.

artemkovalyov reviewed (approved):
For my part, it's good to go.

MatKuhr commented on tutorials/s4sdk-odata-service-cloud-foundry/s4sdk-odata-service-cloud-foundry.md:
```diff
@@ -75,7 +75,7 @@ final List<MyBusinessPartnerType> businessPartners = ODataQueryBuilder
                 "CreationDate")
         .build()
-        .execute(destination)
+        .executeRequest(destination)
```
This change is actually not correct. The `ODataQueryBuilder` is not part of the Cloud SDK and to my knowledge hasn't changed. 

It's not critical because this code is not meant to be run by the reader, but still we should provide valid code. So let's revert this.

MatKuhr commented on tutorials/s4sdk-odata-service-cloud-foundry/s4sdk-odata-service-cloud-foundry.md:
```diff
@@ -107,12 +107,12 @@ final List<BusinessPartner> businessPartners =
                         BusinessPartner.IS_MALE,
                         BusinessPartner.IS_FEMALE,
                         BusinessPartner.CREATION_DATE)
-                .execute(destination);
```
Here the change is correct 👍🏻

MatKuhr reviewed (changes requested):
mostly lgtm, just found one minor mistake in the coding. I limited my review to the semantics of the updates.

MichaelCzcz commented:
@thecodester New tutorials have been added to this PR. Please take over.

thecodester commented:
@Johannes-Schneider This PR now shows a bunch of other commits and changes -- I'm sure they are related to merge commits and will cause any issues. In any event, can we close this PR and you can push again the tutorials you want to push to production. I also suggest first to reset your fork (Tutorial Helper --> Reset Fork) before pushing.
